lightning/src/ln/channel.rs

```diff
@@ -668,6 +668,13 @@ pub(super) struct ChannelContext<Signer: ChannelSigner> {
 	// cost of others, but should really just be changed.
 
 	cur_holder_commitment_transaction_number: u64,
+
+	// The commitment point corresponding to `cur_holder_commitment_transaction_number`, which is the
+	// *next* state. We recompute it each time the state changes because the state changes in places
+	// that might be fallible: in particular, if the commitment point must be fetched from a remote
+	// source, we want to ensure it happens at a point where we can actually fail somewhat gracefully;
+	// i.e., force-closing a channel is better than a panic!
```

Comment:
well, if you can't compute the commitment point, then you can't force-close either.  and once you can force close, the signer is back, and you can just continue as normal instead.  so it doesn't seem like `ChannelClose::Error` is the right action in this case.  crashing may actually be more reasonable until we have a retry mechanism (you'll retry when you restart).

Reply:
:facepalm: Yes, of course... you are right about this.